mailpoet/tests/performance/tests/newsletter-statistics.js

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable import/no-unresolved */
+/* eslint-disable import/no-default-export */
+/**
+ * External dependencies
+ */
+import { sleep, check } from 'k6';
+import { chromium } from 'k6/experimental/browser';
+import { randomIntBetween } from 'https://jslib.k6.io/k6-utils/1.1.0/index.js';
+
+/**
+ * Internal dependencies
+ */
+import {
+  baseURL,
+  thinkTimeMin,
+  thinkTimeMax,
+  headlessSet,
+  timeoutSet,
+} from '../config.js';
+import { authenticate, waitForSelectorToBeVisible } from '../utils/helpers.js';
+
+export async function newsletterStatistics() {
+  const browser = chromium.launch({
+    headless: headlessSet,
+    timeout: timeoutSet,
+  });
+  const page = browser.newPage();
+
+  try {
+    // Go to the page
+    await page.goto(`${baseURL}/wp-admin/admin.php?page=mailpoet-newsletters`, {
+      waitUntil: 'networkidle',
+    });
+
+    // Log in to WP Admin
+    authenticate(page);
+
+    // Wait for async actions and open newsletter statistics
+    await page.waitForNavigation({ waitUntil: 'networkidle' });
+    await page.goto(
+      `${baseURL}/wp-admin/admin.php?page=mailpoet-newsletters#/stats/1`,
+      {
+        waitUntil: 'networkidle',
+      },
+    );
+
+    // Wait for the page to load and click sold tab
+    page.waitForSelector('.mailpoet-listing-table');
+    page.waitForSelector('[data-automation-id="products-sold-tab"]');
+    page.waitForLoadState('networkidle');
+    page.locator('[data-automation-id="products-sold-tab"]').click();
+    page.waitForLoadState('networkidle');
+    await waitForSelectorToBeVisible(
+      page,
+      '.mailpoet-tab-content > p:nth-child(1)',
+    );
+    check(page, {
+      'no products present as sold text is present':
+        page.locator('.mailpoet-tab-content > p:nth-child(1)').textContent() ===
+        'Unfortunately, no products were sold as a result of this email!',
+    });
+
+    // Click the subscribers engagement tab
+    page.locator('[data-automation-id="engagement-tab"]');
+    page.waitForSelector('.mailpoet-listing-table');
+    waitForSelectorToBeVisible(
```

Comment:
All promises will need to be awaited to prevent some race conditions. Aren't all the `waitFor...` functions promises?

Reply:
I was thinking the same, maybe I could add `await` before all `page.` ... What do you think @JanJakes ?

Reply:
@veljkho I guess that could work. Probably TypeScript would help here (knowing which ones are promises), but I don't know how much work that would be, we can think about it for a hacking topic, maybe.

Reply:
I'm planning to make a separate after-project PR to optimize the test suite with all necessary small changes and will include this. Please approve the PR if you don't have any other comments.

Reply:
Hi @JanJakes, I have included this in my latest PR #4820 . Thanks for suggesting it.